### PR TITLE
Anonymizer to receive original_text

### PR DESCRIPTION
**Pull request by SharonHart**
Both mask and hash require the original text for the anonymization.
This changes the interface, so that anonymize method will receive it.

### CODE REVIEW

File: presidio-anonymizer/presidio_anonymizer/anonymizers/hash.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
 """Hashes the PII text entity."""
+from presidio_anonymizer.anonymizers import Anonymizer
 
 
-class Hash(object):
+class Hash(Anonymizer):
```
Comment (SharonHart) [inline presidio-anonymizer/presidio_anonymizer/anonymizers/hash.py:5]: @shiranr this was not intentional right?

File: presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py [hunk 2]
```diff
@@ -30,17 +30,24 @@ def anonymize(self, engine_request: AnonymizerRequest) -> str:
         original_text = engine_request.get_text()
         last_replacement_point = len(original_text)
         output_text = engine_request.get_text()
-        analyzer_results = engine_request.get_analysis_results(
-        ).to_sorted_unique_results(True)
+        analyzer_results = (
```
Comment (itye-msft) [inline presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py:33]: What are the new braces for?

**Review by itye-msft — APPROVED** [review]
Approved with a question

File: presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py (hunk 2 above)
Comment (SharonHart) [reply presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py:33]: Actually I'm not sure, black added them

**Review by navalev — APPROVED** [review]
apprvoing - alignes with our offline discussion

File: presidio-anonymizer/presidio_anonymizer/anonymizers/hash.py (hunk 1 above)
Comment (shiranr) [reply presidio-anonymizer/presidio_anonymizer/anonymizers/hash.py:5]: Nope, an honest mistake. :)

File: presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py [hunk 3]
```diff
@@ -9,6 +9,6 @@ class Anonymizer(ABC):
     logger = logging.getLogger("presidio-anonymizer")
 
     @abstractmethod
-    def anonymize(self, params={}):
+    def anonymize(self, original_text=None, params={}):
```
Comment (shiranr) [inline presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py:12]: Please notice, not all anonymize methods require the original_text.

Comment (shiranr) [reply presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py:12]: The original_text can be passed through params
or better yet we can change params to **kwargs

Comment (SharonHart) [reply presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py:12]: I know, but can we avoid it in the interface level?

**Review by shiranr — DISMISSED** [review]

File: presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py (hunk 3 above)
Comment (shiranr) [reply presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py:12]: yes, use **kwargs instead of params.
params was a temporary place holder

**Comment by SharonHart** [comment]
@shiranr I don't think **kwargs can go on an interface, also params need to be method specific, and the current framework is not supporting appending to it as it comes from transformation straight from the payload. 
What do you suggest?

File: presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py (hunk 3 above)
Comment (SharonHart) [reply presidio-anonymizer/presidio_anonymizer/anonymizers/anonymizer.py:12]: **kwargs essentially let you pass anything so it misses the idea of an interface.
Interface changes between implementations, so it breaks the open close principle

File: presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py (hunk 2 above)
Comment (itye-msft) [reply presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py:33]: Looks unnecessary

**Pull request merged**